Jetcaster/wear/src/main/java/com/example/jetcaster/ui/podcast/PodcastDetailsViewModel.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetcaster.ui.podcast
+
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.jetcaster.core.data.model.PlayerEpisode
+import com.example.jetcaster.core.data.model.asExternalModel
+import com.example.jetcaster.core.data.repository.EpisodeStore
+import com.example.jetcaster.core.data.repository.PodcastStore
+import com.example.jetcaster.core.player.EpisodePlayer
+import com.example.jetcaster.ui.PodcastDetails
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
+
+/**
+ * ViewModel that handles the business logic and screen state of the Podcast details screen.
+ */
+@HiltViewModel
+class PodcastDetailsViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    episodeStore: EpisodeStore,
+    private val episodePlayer: EpisodePlayer,
+    podcastStore: PodcastStore
+) : ViewModel() {
+
+    private val podcastUri: String = savedStateHandle[PodcastDetails.podcastUri]!!
```

Comment:
I wonder about the !!. I do it myself, but is dangerous if you ever enable deeplinking.

Reply:
shall I fail the screen if the podcastUri is not found?

Reply:
It's fine for a sample. I think generally being able to crash an app is a bad thing.  So yeah, Ideally log the error and return to home screen?

Reply:
What about the approach in the tv module: https://github.com/android/compose-samples/blob/jetcaster/all_form_factors/Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/podcast/PodcastScreenViewModel.kt#L49
It checks if the URI is null and return an empty/error screen instead

Reply:
done